File: Math/water_bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        ans = numBottles
        while numBottles >= numExchange:
            rem, div = numBottles % numExchange, numBottles // numExchange
            ans += div
            numBottles = rem + div
        return ans

File: Math/CPP/water_bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        int ans = numBottles;
        while (numBottles >= numExchange) {
            int div = numBottles / numExchange;
            int rem = (numBottles % numExchange);
            numBottles = div + rem;
            ans += div;
        }
        return ans;
    }
};